one_fm/purchase/utils.py

```diff
@@ -114,3 +114,9 @@ def filter_description_specific_for_item_group(doctype, txt, searchfield, start,
         return frappe.db.sql(query.format(filters.get("doctype")),
             {'start': start, 'page_len': page_len, 'txt': "%%%s%%" % txt}
         )
+
+@frappe.whitelist()
+def check_for_signature(doc, method):
+	if doc.status == "Completed":
+		if not doc.authority_signature:
+			frappe.throw(__('Please Sign the form to Accept the Request'))
```

Comment:
There is an approval workflow for Purchase Order and Purchase Receipt and the signature needs to be checked on the approval only
`if doc.workflow_sate == 'Approved' and not doc.signature_field: throw('Your message')`

Reply:
Got it!

Reply:
@pjamsheer , Done.